ddtrace/utils/formats.py

```diff
@@ -97,7 +98,25 @@ def parse_tags_str(tags_str):
     if not tags_str:
         return parsed_tags
 
-    for tag in tags_str.split(","):
+    # Identify separator based on which successfully identifies the correct
+    # number of valid tags
+    numtagseps = tags_str.count(":")
+    for sep in [",", " "]:
+        if sum(1 for _ in tags_str.split(sep) if _.count(":") == 1) == numtagseps:
```

Comment:
This is so totally my fault, you had the logic right before my last rushed comments! I'm making this final suggestion, but feel free to ignore.
```suggestion
        if sum(":" in _ for _ in tags_str.split(sep)) == numtagseps:
```